### PR TITLE
Feat: Add support for search highlighting

### PR DESCRIPTION
**Pull request by SWH-Relewise**
https://trello.com/c/hfWTsKst/8562-js-sdk-update-with-new-features-from-release-11910

### CODE REVIEW

File: packages/client/src/builders/search/contentHighlightingBuilder.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    private highlightable: ContentHighlightProps = {
+        $type: 'Relewise.Client.Requests.Shared.Highlighting.ContentHighlightProps, Relewise.Client',
+        displayName: false
+    }; 
+    private limit: HighlightSettings2ContentContentHighlightPropsHighlightSettings2Limits = {};
+    private shape: HighlightSettings2ContentContentHighlightPropsHighlightSettings2ResponseShape = {
+        includeOffsets: false
+    };
+
+    public enable(enabled: boolean): this {
```
Comment (mzanoni) [inline packages/client/src/builders/search/contentHighlightingBuilder.ts:14]: Would like the public method to be called `enabled(true)` or `enabled(false)` as this accepts a bools, it reads better

File: packages/client/src/builders/search/contentHighlightingBuilder.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    }
+
+    public build(): ContentSearchSettingsHighlightSettings {
+        return {
+            $type: 'Relewise.Client.Requests.Search.Settings.ContentSearchSettings+HighlightSettings, Relewise.Client',
+            enabled: this.enabled,
+            highlightable: this.highlightable,
+            limit: this.limit,
+            shape: this.shape
+        } as ContentSearchSettingsHighlightSettings;
```
Comment (mzanoni) [inline packages/client/src/builders/search/contentHighlightingBuilder.ts:48]: Would do we need to cast here?

File: packages/client/src/builders/search/productHighlightingBuilder.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    }
+
+    public build(): ProductSearchSettingsHighlightSettings {
+        return {
+            $type: 'Relewise.Client.Requests.Search.Settings.ProductSearchSettings+HighlightSettings, Relewise.Client',
+            enabled: this.enabled,
+            highlightable: this.highlightable,
+            limit: this.limit,
+            shape: this.shape
+        } as ProductSearchSettingsHighlightSettings;
```
Comment (mzanoni) [inline packages/client/src/builders/search/productHighlightingBuilder.ts:48]: Would do we need to cast here?

File: packages/client/src/builders/search/productHighlightingBuilder.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
+import { HighlightSettings2ProductProductHighlightPropsHighlightSettings2Limits, HighlightSettings2ProductProductHighlightPropsHighlightSettings2ResponseShape, ProductHighlightProps, ProductSearchSettingsHighlightSettings } from '../../models/data-contracts';
+
+export class ProductHighlightingBuilder {
```
Comment (mzanoni) [inline packages/client/src/builders/search/productHighlightingBuilder.ts:3]: Could we have had a base builder to unify some of the common code? Like the enabled and shape could have been part of a base builder

**Review by mzanoni — COMMENTED** [review]
Really nice work! 💪 

We are generating the release messages based on PR titles, this PR title is not very user friendly. Could we rename it to the features it actually brings instead?

**Review by mzanoni — CHANGES_REQUESTED** [review]
Chose the wrong mode

File: packages/client/src/builders/search/contentHighlightingBuilder.ts (hunk 1 above)
Comment (SWH-Relewise) [reply packages/client/src/builders/search/contentHighlightingBuilder.ts]: Check!

File: packages/client/src/builders/search/contentHighlightingBuilder.ts (hunk 2 above)
Comment (SWH-Relewise) [reply packages/client/src/builders/search/contentHighlightingBuilder.ts]: No i removed it! 😊

File: packages/client/src/builders/search/productHighlightingBuilder.ts (hunk 3 above)
Comment (SWH-Relewise) [reply packages/client/src/builders/search/productHighlightingBuilder.ts]: No i removed it! 😊

File: packages/client/src/builders/search/productHighlightingBuilder.ts (hunk 4 above)
Comment (SWH-Relewise) [reply packages/client/src/builders/search/productHighlightingBuilder.ts:3]: The shape and limit are its own unique type for either product or content. I could not find a good way to consolidate them. 😶

**Pull request merged**